### DISCUSSION TITLE
How can i get the response headers on the client side?

### DISCUSSION THREAD

**Discussion by liulib** [nuxt/nuxt#16271] [Questions]
### Environment

Nuxt project info:                                                                                            12:35:52

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.17.1`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.15.0`
- Builder:          `vite`
- User Config:      `experimental`, `app`, `modules`, `intlify`, `css`, `unocss`, `build`
- Runtime Modules:  `@pinia/nuxt@0.4.3`, `@intlify/nuxt3@0.2.4`, `@unocss/nuxt@0.46.5`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-zvb66h?file=pages/index.vue

### Describe the bug

hey, sorry for the trouble. I'm new guy to the SSR.

I have tried both of these ways.

### first : 
in this way,headers awalys be empty object

client vue file
```
  // not work
  const url = 'http://www.vvvv.plus/api/article/getTopicList';
  const res = await $fetch.raw(url, {
    method: 'get',
  });

  // how can i get headers here ?
  console.log(res.headers);
```

### second 
i can get headers in the server side, but i don't know how to pass it to the client side.

I have check those discussions, but i can't find the answer.

this doc just dispeared https://github.com/nuxt/framework/discussions/1062

this doc is so simple,i don't konw how to use it https://nuxt.com/docs/api/composables/use-request-headers#userequestheaders

server side ts file

```
export default defineEventHandler(async (event) => {
  const response: any = await $fetch.raw(
    'http://www.vvvv.plus/api/article/getTopicList',
    {
      method: 'get',
    }
  );
  // we can get headers here
  console.log(response.headers);

  return response._data;
});
```

client vue file
```
const url = '/api/test';
const res = await useFetch(url);

// but here is empty
console.log(res.headers);
```

**Comment by danielroe** [comment]
See https://nuxt.com/docs/getting-started/data-fetching#example-pass-cookies-from-server-side-api-calls-on-ssr-response.

**Reply by liulib** [reply]
Thank for your reply, danielroe.

I have tested this scheme before you replied, but i don't know how to get headers in client.

server file

```
for (const header of ['set-cookie', 'cache-control']) {
        if (response.headers.has(header)) {
            appendHeader(event, header, response.headers.get(header));
        }
    }
```


Then i try to get response.headers like this
```
  onResponse({ request, response, options }) {
        console.log(response.headers);
        return response._data
      },
    })
```

or this
```
const headers = useRequestHeaders()
console.log(headers);
```


Both empty object

**Reply by danielroe** [reply]
What do you mean by getting the headers on the client? What are you trying to do?

**Reply by liulib** [reply]
I want to read headers in vue file.

In my /server/api/test.get.ts, I did this
```
for (const header of ['set-cookie', 'cache-control']) {
        if (response.headers.has(header)) {
            appendHeader(event, header, response.headers.get(header));
        }
    }
```

In test.vue, I did this
```
const url = '/api/test'
const res = await useFetch(url)

// How can i read headers in here 
```

If I use axios, I can read headers like this
```
const test = await axios.get(url)
console.log(test.headers);
```

**Comment by Chengyunlai** [comment]
不知道你解决了吗，我的朋友

**Comment by Tarikul-Islam-Tuhin** [comment]
[liulib](https://github.com/liulib)
The `Headers` instance returned by the fetch function is iterable, and some values—like the headers themselves—aren’t immediately visible in the console. To view them, you need to iterate over them, like this:

```
for(let entry of res.headers.entries()) {
    console.log(entry);
}
```
